libs/agency/feature/admin/src/lib/components/transaction-definitions-form/transaction-definitions-form.component.ts

```diff
@@ -218,8 +218,8 @@ export class TransactionDefinitionsFormComponent {
       .getTransactionDefinitionByKey$(this.formGroup.value.key)
       .pipe(
         tap(transactionDefinition => {
-          if (transactionDefinition || !this.formGroup.valid) {
-            this.formGroup.controls['key'].setErrors({ keyExists: true });
+          if (this.formGroup.value.key && transactionDefinition) this.formGroup.controls['key'].setErrors({ keyExists: true });
```

Comment:
Can you put this in curly brackets?

if you are just returning, 1 line is ok but if you are handling any logic it is more readable and updatable in curly brackets
```suggestion
          if (this.formGroup.value.key && transactionDefinition) {
            this.formGroup.controls['key'].setErrors({ keyExists: true });
          }
```